connection.go

```diff
@@ -87,11 +87,12 @@ type connection struct {
 	sentHaves        bitmap.Bitmap
 
 	// Stuff controlled by the remote peer.
-	PeerID             PeerID
-	PeerInterested     bool
-	PeerChoked         bool
-	PeerRequests       map[request]struct{}
-	PeerExtensionBytes pp.PeerExtensionBits
+	PeerID                PeerID
+	PeerInterested        bool
+	PeerChoked            bool
+	PeerRequests          map[request]struct{}
+	PeerExtensionBytes    pp.PeerExtensionBits
+	PeerPrefersEncryption bool // as indicated by 'e' field in extension handshake
```

Comment:
I'm guessing you've checked this isn't recorded elsewhere. Currently I guess it's not come up that we may want to remember the prefer encryption flag, we just use it in selecting peers and then forget about it, but now we're propagating this information at a point after connecting?

Reply:
Well, the preference is communicated in the extended handshake, and the PEX spec tells us to pass the word so they can better decide whom to dial. So indeed we keep that indication around just for sharing with peers, and by far it was useless to keep, but that's a fairly low overhead. And we, as receivers of said flags, may also want to match peers against our preference when picking a peer to contact next.